public/client/views/MessagesView.js

```diff
@@ -24,10 +25,32 @@ var MessagesView = Backbone.View.extend({
     //message.cid is unique client-only id
     if (!this.onscreenMessages[message.cid]) {
       var messageView = new MessageView ({model : message});
+      this.renderUsername(message);
       this.$el.append (messageView.render());
       this.onscreenMessages[message.cid] = true;
       $('#messagesView').scrollTop(2000);
     }
+  },
+
+  renderUsername: function(message) {
+    if (!this.currentRoom || this.currentRoom === message.attributes.room) {
```

Comment:
can you please memoize `message.attributes` since you are performing that lookup 7 times in this method?